twake/backend/node/src/core/platform/services/tracker/index.ts

```diff
@@ -0,0 +1,39 @@
+import { Consumes, TwakeService } from "../../framework";
+import TrackerAPI, { IdentifyObjectType, TrackedEventType } from "./provider";
+import Analytics from "analytics-node";
+
+@Consumes([])
+export default class Tracker extends TwakeService<TrackerAPI> implements TrackerAPI {
+  name = "tracker";
+  version = "1";
+  analytics: Analytics;
+
+  public async identify(
+    identity: IdentifyObjectType,
+    callback?: (err: Error) => void,
+  ): Promise<Analytics> {
+    if ((await this.getAnalytics("segment")) && identity)
+      return (await this.getAnalytics("segment")).identify(identity, callback);
+  }
+
+  public async track(
+    tracker: TrackedEventType,
+    callback?: (err: Error) => void,
+  ): Promise<Analytics> {
+    if ((await this.getAnalytics("segment")) && tracker)
+      return (await this.getAnalytics("segment")).track(tracker, callback);
```

Comment:
You await the same thing too many times, in this function and above. Please refactor

Reply:
done 👍🏻